core/src/gl/texture.h

```diff
@@ -107,9 +106,19 @@ class Texture {
 
     bool sanityCheck(size_t _width, size_t _height, size_t _bytesPerPixel, size_t _length) const;
 
+    void freeBufferData() {
+        std::free(m_buffer);
+        m_buffer = nullptr;
+    }
+    void setBufferData(GLubyte* buffer, size_t size) {
+        if (m_buffer == buffer) { return; }
+        std::free(m_buffer);
+        m_buffer = buffer;
+    }
+
     TextureOptions m_options;
 
-    std::unique_ptr<GLubyte> m_buffer;
+    GLubyte* m_buffer = nullptr;
```

Comment:
So unique_ptr doesn't free the pointer it holds? I assumed the default deleter would call free - does it just destruct the pointer itself?

Reply:
unique_ptr default deleter is calling delete on the malloc'd pointer which Valgrind was so nice to point out.

Reply:
Oh what about clang static analyzer? Any experiences with it? Would probably be good to have it in our before merging bigger changes checklist as well

Reply:
yeah static analyzer should have caught something like this.

Reply:
I was curious so I ran the xcode static analyzer with the old version of this - no warning about the leak!

Reply:
To be a little more careful, and maybe help ourselves in other situations, we could make a struct to wrap malloced memory that frees it in a destructor.

Reply:
Yeah RAII construct will definitely be better here. Something like this: https://embeddedartistry.com/blog/2017/2/23/c-smart-pointers-with-aligned-mallocfree

Reply:
Another more concise of what the above link shows: https://gist.github.com/evakili/a1b9fd65c496046dcd24

Reply:
Yeah using a custom unique_ptr this way sounds good. We can add it now or later.